File: blacklist/contracts-java/src/test/java/net/corda/examples/attachments/states/StateTests.java
package net.corda.examples.attachments.states;

import net.corda.core.contracts.ContractState;
import net.corda.core.identity.CordaX500Name;
import net.corda.core.identity.Party;
import net.corda.testing.core.TestIdentity;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class StateTests {
    private final Party alice = new TestIdentity(new CordaX500Name("Alice", "", "GB")).getParty();
    private final Party bob = new TestIdentity(new CordaX500Name("Bob", "", "GB")).getParty();

    @Test
    public void agreementStateHasParamsOfCorrectTypeInConstructor() {
        new AgreementState(alice, bob, "this is an agreement txt");
    }

    @Test
    public void tokenStateHasGettersForPartyAPartyBandTxt() {
        AgreementState agreementState = new AgreementState(alice, bob, "this is an agreement txt");
        assertEquals(alice, agreementState.getPartyA());
        assertEquals(bob, agreementState.getPartyB());
        assertEquals("this is an agreement txt", agreementState.getTxt());
    }

    @Test
    public void agreementStateImplementsContractState() {
        assertTrue(new AgreementState(alice, bob, "this is an agreement txt") instanceof ContractState);
    }

    @Test
    public void agreementStateHasTwoParticipantsPartyAPartyB() {
        AgreementState agreementState = new AgreementState(alice, bob, "this is an agreement txt");
        assertEquals(2, agreementState.getParticipants().size());
        assertTrue(agreementState.getParticipants().contains(alice));
        assertTrue(agreementState.getParticipants().contains(bob));
    }
}

File: blacklist/contracts-kotlin/src/test/kotlin/net/corda/examples/attachments/states/StateTests.kt
package net.corda.examples.attachments.states

import net.corda.core.identity.CordaX500Name
import net.corda.testing.core.TestIdentity
import org.junit.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class StateTests {

    private val alice = TestIdentity(CordaX500Name("Alice", "", "GB")).party
    private val bob = TestIdentity(CordaX500Name("Bob", "", "GB")).party

    @Test
    fun `agreementStateHasParamsOfCorrectTypeInConstructor`() {
        AgreementState(alice, bob, "this is an agreement txt")
    }

    @Test
    fun `agreementStateHasGettersForPartyAPartyBTxt`() {
        val agreementState = AgreementState(alice, bob, "this is an agreement txt")
        assertEquals(alice, agreementState.partyA)
        assertEquals(bob, agreementState.partyB)
        assertEquals("this is an agreement txt", agreementState.txt)
    }

    @Test
    fun `agreementStateHasTwoParticipantsPartyAPartyB`() {
        val agreementState = AgreementState(alice, bob, "this is an agreement txt")
        assertEquals(2, agreementState.participants.size)
        assertTrue(agreementState.participants.contains(alice))
        assertTrue(agreementState.participants.contains(bob))
    }
}